modules/clairs/1.0/clairs.smk

```diff
@@ -0,0 +1,307 @@
+                        {params.options}
+                "
+                > {log.stdout} 2> {log.stderr} 
+        """)
+
+
+    # Annotates VCF file with gnomAD frequency data and filters out poor calls.
+rule _clairs_gnomad_annotation:
+    input:
+        vcf = str(rules._clairs_call_variants.output.vcf),
+        tbi = str(rules._clairs_call_variants.output.tbi),
+        gnomad = reference_files("genomes/{genome_build}/variation/af-only-gnomad.{genome_build}.vcf.gz")
+    output:
+        vcf = temp(CFG["dirs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/output.gnomad.vcf.gz"),
+        tbi = temp(CFG["dirs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/output.gnomad.vcf.gz.tbi")
+    log:
+        stderr = CFG["logs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/clairs_gnomad_annotation.stderr.log",
+        stdout = CFG["logs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/clairs_gnomad_annotation.stdout.log"
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    threads:
+        CFG["threads"]["bcftools"]
+    resources:
+        **CFG["resources"]["gnomad"]
+    shell:
+        op.as_one_line("""
+        bcftools annotate --threads {threads}
+        -a {input.gnomad} -c INFO/AF {input.vcf} |
+        awk 'BEGIN {{FS=OFS="\\t"}} {{ if ($1 !~ /^#/ && $8 !~ ";AF=") $8=$8";AF=0"; print $0; }}' |
+        bcftools view -i 'FILTER="PASS" && INFO/AF < 0.0001 && FMT/DP[0] >= 8 && FMT/AF[0] >= 0.1 && (FMT/NDP[0] < 10 || FMT/NAF[0] < 0.3)'
+        -Oz -o {output.vcf} 2> {log.stderr}
+        &&
+        tabix -p vcf {output.vcf} 2>> {log.stderr}
+        """)
+
+
+# Filters out PoN variants
+rule _clairs_filter_pon:
+    input:
+        vcf = str(rules._clairs_gnomad_annotation.output.vcf), 
+        tbi = str(rules._clairs_gnomad_annotation.output.tbi),
+        pon = reference_files("genomes/{genome_build}/ont/colorsDb.v1.2.0.deepvariant.glnexus.{genome_build}.vcf.gz")
+    output:
+        vcf = CFG["dirs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/clairs.final.vcf.gz",
+        tbi = CFG["dirs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/clairs.final.vcf.gz.tbi"
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    resources:
+        **CFG["resources"]["filter"]
+    threads:
+        CFG["threads"]["bcftools"]
+    log:
+        stderr = CFG["logs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/filter_pon.stderr.log",
+        stdout = CFG["logs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/filter_pon.stdout.log"
+    shell:
+        op.as_one_line("""
+        bcftools isec -C -w1 -O z -o {output.vcf} {input.vcf} {input.pon} > {log.stdout} 2> {log.stderr}
+        &&
+        tabix -p vcf {output.vcf} >> {log.stdout} 2>> {log.stderr}
+        """)
+
```

Comment:
Normally here would be some sort of vcf filtering for depth, number of reads etc or at least the variants with PASS flag. I don't think I realize where this step is done in this snakefile. I might have missed where the vcf filtering is happening besides this pon rule - or is it not happening?